src/shared/layouts/BaseLayout/components/TopBar/Notifications/Notification/Notification.tsx

```diff
@@ -25,67 +26,116 @@ import {
   StyledTopSection,
   StyledBottomSection,
   StyledBtn,
+  StyledLogoPlug,
 } from './Notification.styles';
 import { NotificationProps } from './Notification.types';
 
 export const Notification = ({
   currentId,
   setCurrentId,
-  accountId,
   alertId,
-  label,
-  title,
+  workspaceName,
+  appletId,
+  appletName,
+  appletImage,
+  respondentSecretId,
   message,
-  imageSrc,
   timeAgo,
-  viewed,
+  isWatched,
+  respondentId,
   encryption,
-  appletId,
+  alert,
 }: NotificationProps) => {
   const { t } = useTranslation('app');
   const dispatch = useAppDispatch();
   const isActive = currentId === alertId;
   const [passwordPopupVisible, setPasswordPopupVisible] = useState(false);
+  const navigate = useNavigate();
+  const { getAppletPrivateKey } = useEncryptionStorage();
+  const hasEncryptionCheck = !!getAppletPrivateKey(appletId ?? '');
 
   const handleNotificationClick = async () => {
-    const { updateAlertStatus } = account.thunk;
+    const { setAlertWatched } = alerts.thunk;
     setCurrentId(isActive ? '' : alertId);
-    const result = await dispatch(updateAlertStatus({ alertId }));
 
-    if (updateAlertStatus.fulfilled.match(result)) {
-      await dispatch(account.thunk.switchAccount({ accountId }));
-    }
+    if (isWatched) return;
+
+    await dispatch(
```

Comment:
maybe it makes sense to call dispatch(setAlertWatched) and then if the action was fulfilled call the updateAlertWatchedState action? to avoid blinking and unnecessary state updates?